### PR TITLE
JBPM-8244 'No server connected' warning is displayed when more server…

### PR DESCRIPTION
**Pull request by bxf12315**
…s are connected

@cristianonicolai Please review it.

### CODE REVIEW

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

File: jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilder.java [hunk 1]
```diff
@@ -100,6 +100,11 @@ protected void loadServerTemplates() {
                 } else {
                     widgetView.clearSelectedServerTemplate();
                     view.clearSelectedServerTemplate();
+                    if(ids.size() > 1) {
```
Comment (nmirasch) [inline jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilder.java]: @bxf12315 missing space after if

File: jbpm-wb-common/jbpm-wb-common-client/src/test/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilderTest.java [hunk 2]
```diff
@@ -248,4 +249,100 @@ public void testServerTemplateChangeHandler(){
         inOrder.verify(viewWidget).updateSelectedValue(serverTemplateId);
         inOrder.verify(serverTemplateSelectedEvent).fire(any());
     }
-}
\ No newline at end of file
+
+    @Test
+    public void testLoadServerTemplatesForSelectedServerTemplateIsNull(){
```
Comment (nmirasch) [inline jbpm-wb-common/jbpm-wb-common-client/src/test/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilderTest.java]: @bxf12315 missing space before '{'

File: jbpm-wb-common/jbpm-wb-common-client/src/test/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilderTest.java [hunk 3]
```diff
@@ -248,4 +249,100 @@ public void testServerTemplateChangeHandler(){
...
+        final String serverTemplateId1 = "id1";
+        final ServerTemplate st1 = new ServerTemplate(serverTemplateId1,
```
Comment (nmirasch) [inline jbpm-wb-common/jbpm-wb-common-client/src/test/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilderTest.java]: @bxf12315 This is not corrected by 'Reformat code' following our code conventions but we agreed to keep the parameters at the same line if it's possible. Maybe you can follow the same idea in this whole test. The point is  incrementally update the code we need to change for other reasons, avoiding to do a big update just for reformat the code.

File: jbpm-wb-common/jbpm-wb-common-client/src/test/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilderTest.java [hunk 4]
```diff
@@ -248,4 +249,100 @@ public void testServerTemplateChangeHandler(){
...
+        verify(view, times(2)).addServerTemplate(any());
+        verify(view).removeAllServerTemplates();
+        verify(view).clearSelectedServerTemplate();
+        verify(viewWidget).selectServerTemplate(serverTemplateId1);
+        verify(view).selectServerTemplate(serverTemplateId1);
+        verify(viewWidget).setVisible(true);
+        verify(view).setVisible(true);
+
+        verifyNoMoreInteractions(view);
+
```
Comment (nmirasch) [inline jbpm-wb-common/jbpm-wb-common-client/src/test/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilderTest.java]: @bxf12315 extra line at the end not needed

**Comment by nmirasch** [comment]
@bxf12315 Looks good to me,  I just  added some minor comments regarding the code formatting but the code works fine.

File: jbpm-wb-common/jbpm-wb-common-client/src/test/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilderTest.java [hunk 5]
```diff
@@ -248,4 +249,100 @@ public void testServerTemplateChangeHandler(){
...
+        verify(view).selectServerTemplate(serverTemplateId1);
+
+        verify(viewWidget).setVisible(true);
+        verify(view).setVisible(true);
+
+        verifyNoMoreInteractions(view);
+    }
+
+    @Test
+    public void testLoadServerTemplatesForSelectedServerTemplateNotNullAndServerTemplateListNotContainsIt(){
```
Comment (nmirasch) [inline jbpm-wb-common/jbpm-wb-common-client/src/test/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilderTest.java]: @bxf12315 the longer test name I have seen ever but it explain really well what is it doing. ;)

**Comment by bxf12315** [comment]
@cristianonicolai @nmirasch updated, Please review again.

File: jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilder.java [hunk 6]
```diff
@@ -88,7 +88,7 @@ protected void loadServerTemplates() {
                 widgetView.addServerTemplate(id);
                 view.addServerTemplate(id);
             }
-
+
```
Comment (jstastny-cz) [inline jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilder.java]: remove trailing whitespace

File: jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilder.java [hunk 7]
```diff
@@ -100,6 +100,11 @@ protected void loadServerTemplates() {
                 } else {
                     widgetView.clearSelectedServerTemplate();
                     view.clearSelectedServerTemplate();
+                    if(ids.size() > 1) {
+                        String defaultServerTemple = ids.get(0);
```
Comment (jstastny-cz) [inline jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilder.java]: Temple -> Template .... I wouldn't try to save 2 letters in variable name here

**Comment by bxf12315** [comment]
@jstastny-cz updated

File: jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilder.java (hunk 6 above)
Comment (cristianonicolai) [reply jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilder.java]: @bxf12315 really minor but can you fix this before merging?

Comment (bxf12315) [reply jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/menu/ServerTemplateSelectorMenuBuilder.java]: @cristianonicolai updated.

**Pull request merged**